src/lib/pickles/compile.ml

```diff
@@ -243,6 +248,37 @@ module Storables = struct
     }
 end
 
+let create_lock () =
```

Comment:
This should never have made it into the codebase. I am annoyed.

Reply:
Disagree. It is only applied to sequencing the circuit runs that `compile` creates when compiling different step constraint systems. This is necessary and has none of the downsides we discussed